### ISSUE TITLE
CNTK doesn't support Noise layer?

### ISSUE THREAD

**Issue by ml-pickle** [keras-team/keras#8972]
If you try to use a Gaussian noise layer I get a `randomness op with dynamic shape is not supported now.` error.

Doesn't matter if input shape is (batch, feats) or (batch, seq_len, feats).

```
X = np.random.random((5, 10))
y = np.array([0,1,0,1,0])
input_ = Input(shape=(10,))
input_noised = GaussianNoise(1e-6)(input_)
h_o = Dense(5)(input_noised)
p = Dense(1, activation='sigmoid')(h_o)
model = Model(inputs=input_sentence, outputs=p)
model.compile(loss='binary_crossentropy', optimizer='adam')
model.fit(X,y, epochs=10, batch_size=2)
```

https://github.com/keras-team/keras/blob/cb6adf8da72c2d685b3a0a192165949c9da7b0e7/keras/backend/cntk_backend.py#L440-L450

I guess the batch being a dynamic axis makes it impossible to use a noise layer?  Seems strange.  Thoughts @souptc ?

**Comment by souptc** [comment]
CNTK‘s philosophy is trying to hidden the batch dimension from user, so currently we don't have an user api that could generate random samples across batch axis.

CNTK do have other way to do noise layer, but that will introduce much more changes in keras layer code and can't wrapped those in cntk_backend, so we will prefer to fix the random_normal api.

We have this task traced in backlog,  but unfortunately, we haven't finish this...
